### PR TITLE
Add Scalingo builds

### PR DESCRIPTION
**Pull request by estellecomment**
Sister PR (branch with same name) : https://github.com/tchapgouv/matrix-react-sdk/pull/371
Scalingo machine : tchap-web-staging

---
This PR enables to build and run tchap-web on scalingo, with all deps in package.json (no yarn links). matrix-react-sdk is specified from a github branch, and matrix-js-sdk as a "standard" package.

~~Out of scope : build with matrix-js-sdk from a github branch. Because this is only needed for v2, and hopefully not v4.~~
v2 needs a github branch version for matrix-js-sdk, because of security patches we added. (v4 builds completely differently anyway, so this PR will probably not be reused for v4)

Note : the scalingo env var NPM_CONFIG_PRODUCTION=false is needed so that Scalingo installs the devdependencies. We need the devdependencies in matrix-react-sdk to build it.
We could try to find something smarter where the devdependencies of tchap-web are not installed...

Todo 
 - [x] cleanup : instead of renaming the "build" command and editing the "package.sh" script, make separate commands and configure scalingo to run them (right now "build" is the default so scalingo runs it) 
 - [x] NPM_CONFIG_PRODUCTION, see above. If we can't get rid of it, it would be better to place it in a config file, and document it.
-> replaced by a `production=false` flag in the appropriate build command.
 - [x] I also removed the "prepare" script from matrix-react-sdk, because it was automatically run after yarn install, and crashed. It crashed because of the `git rev-parse` command : `fatal: not a git repository (or any of the parent directories): .git`. The easiest would be to replace that `git rev-parse` with a script which fetches the version from package.json, the way it is done in `scripts/package.sh`.
-> Fixed : prepare is actually not run for the subpackage matrix-react-sdk, juste for tchap-web package. So we dodge that problem :)
  - [x] if matrix-react-sdk is a standard version instead of a pointer to a git branch, does the whole thing still work ? -> looks like it. We don't use this case anyway since we don't do npm releases (for now)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,8 +78,8 @@
     "gemini-scrollbar": "github:matrix-org/gemini-scrollbar#b302279",
     "gfm.css": "^1.1.2",
     "highlight.js": "^9.13.1",
-    "matrix-js-sdk": "1.0.4",
-    "matrix-react-sdk": "1.0.7",
+    "matrix-js-sdk": "2.2.0",
```
Comment (jdauphant) [inline package.json]: :thinking: ça va utiliser des versions différentes du JS sdk sur react-sdk et ici ?

Comment (estellecomment) [reply package.json]: faut faire gaffe de les garder in sync, oui... Je sais pas dans le detail pourquoi ya un import ici aussi.

**Comment by estellecomment** [comment]
Ready for review @jdauphant

File: package.json (hunk 1 above)
Comment (jdauphant) [reply package.json]: Il y a une faille de sécurité dans le js-sdk qui a été corrigé dans la version patché de Leo. Ce n'est pas le cas dans la 2.2.0

Comment (jdauphant) [reply package.json]: On peut merge et régler ce problème avant mise en beta

**Review by jdauphant — APPROVED** [review]
Modulo le problème de sdk JS (faille de sécurité qui nécessite la dernière version de olm), on peut merge et avoir en tête de ne pas oublier cette faille.

**Comment by estellecomment** [comment]
Hmm, du coup on doit dépendre d'un js-sdk d'une branche github, j'avais oublié ce détail.
Ok, je regarde ca.

**Pull request merged**